markets/storageadapter/client.go

```diff
@@ -368,7 +368,7 @@ func (c *ClientNodeAdapter) OnDealExpiredOrSlashed(ctx context.Context, dealID a
 	return nil
 }
 
-func (c *ClientNodeAdapter) SignProposal(ctx context.Context, signer address.Address, proposal market0.DealProposal) (*marketactor.ClientDealProposal, error) {
```

Comment:
I'm assuming that at least for valid string labels cbor serialization stays the same, so we're not worrying about invalid signatures with current actors?

Reply:
Correct.  And if I understand you right that's the entire point of all this union type hoop-jumping.